### ISSUE TITLE
Docker builder with powershell provisioner always returns 0

### ISSUE THREAD

**Issue by pintonunes** [hashicorp/packer#7621]
Packer version 1.4.0
Building Windows containers

When running the powershell provisioner with docker the powershell exit code is not reported back to packer. Is always 0

The issue is quite simple. The execute_command already has the powershell.exe. 
When running in the docker provisioner, the execute_command runs already in a powershell console.

![image](https://user-images.githubusercontent.com/37883272/57365532-22c99e00-717d-11e9-8731-dcbc5763bd89.png)

Workaround:
"execute_command":  "\u0026 { if (Test-Path variable:global:ProgressPreference){$ProgressPreference=\"SilentlyContinue\"};. {{.Vars}}; \u0026\"{{.Path}}\"; exit $LastExitCode }"

Basically, removing the "powershell -executionpolicy bypass " from the execute command solves this.

**Comment by SwampDragons** [comment]
Changing this so that it doesn't call Powershell inside the provisioner will break things for builders that aren't docker. 

What happens if you change your docker builder run_command to `["-d", "-i", "-t", "{{.Image}}"]`?

**Comment by pintonunes** [comment]
Let me try that. Will let you know.. 
And change the entrypoint to cmd.exe? This would make the provisioner to run like the other builders (But then will break the other provisioners like the "file" one )

**Comment by azr** [comment]
Hey @pintonunes, this could have been fixed by https://github.com/hashicorp/packer/pull/7657, I recently released a nightly build of this, can you please tell us wether this fixed it for you ?

https://github.com/hashicorp/packer/releases/tag/nightly

**Comment by azr** [comment]
Hello here, I added the possibility to set the "execution_policy": "none" per powershell provisioner in this PR: https://github.com/hashicorp/packer/pull/7678

This way, the exit code is returned ! 👍

```
{
  "builders": [
    {
      "type": "docker",
      "image": "microsoft/windowsservercore:1709",
      "windows_container": true,
      "commit": true
    }
  ],
  "provisioners": [
    {
      "type": "powershell",
      "inline": ["dir c:\\", "exit(32)"],
      "execution_policy": "none"
    }
  ]
}
-> docker: Script exited with non-zero exit status: 32.Allowed exit codes are: [0]
```

Binaries can be found here: https://circleci.com/gh/hashicorp/packer/2752#artifacts/containers/0

Does it seems like a valid solution to you @SwampDragons & @pintonunes ?

**Comment by SwampDragons** [comment]
I think this is a great solution.

**Comment by azr** [comment]
Yay, awesome ! :D  Closing this issue as #7678 got merged.
